app/components/preprint-form-authors.js

```diff
@@ -301,11 +243,19 @@ export default CpPanelBodyComponent.extend(Analytics, {
         },
     },
     /* If user removes their own admin permissions, many things on the page must become
-    disabled.  Changing the isAdmin flag to false will remove many of the options
-    on the page. */
-    removedSelfAsAdmin(contributor, permission) {
-        if (this.get('currentUser').id === contributor.get('userId') && permission !== 'ADMIN') {
-            this.set('isAdmin', false);
+    disabled.  Changing the isAdmin flag to false will remove many options on the
+    contributor section. If the user is downgraded to write, they can still edit
+    most of the page, but cannot update authors */
+    modifiedSelf(contributor, permission, removed) {
```

Comment:
Now that read and write contributors can both access the Edit preprint page, and Admins can remove themselves from preprints on edit, we need more logic for where Admins are removing or demoting their permissions, or read/write are removing themselves.  The UI needs to change instantly to reflect their new permissions.

Reply:
so maybe a confirmation modal when anyone tries to remove themselves?

Reply:
Yeah I think that makes sense. That can probably wait or be deferred if it is at all complicated

Reply:
okay, have a confirmation modal
<img width="1082" alt="screen shot 2018-10-03 at 8 32 45 pm" src="https://user-images.githubusercontent.com/9755598/46448399-4098d580-c74c-11e8-85b3-1e91b8cc07ce.png">